src/snowflake/cli/_plugins/nativeapp/entities/models/event_sharing_telemetry.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Optional
+
+from click import ClickException
+from pydantic import Field, field_validator, model_validator
+from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
+
+
+class EventSharingTelemetry(UpdatableModel):
+    authorize_event_sharing: Optional[bool] = Field(
+        title="Whether to authorize Snowflake to share application usage data with application package provider. This automatically enables the sharing of required telemetry events.",
+        default=None,
+    )
+    optional_shared_events: Optional[List[str]] = Field(
```

Comment:
do these need to be optional to have a default? do we handle these fields differently if they're `None` vs `False`/`[]`?

Reply:
i am not sure if optional [] as default will have any side effect (I know it's not recommended in function params)